### ISSUE TITLE
Create trimming utils to implement different kind of trimming

### ISSUE THREAD

**Issue by magicDGS** [magicDGS/ReadTools#26]
HTSJDK have its own [`TrimmingUtils`](https://github.com/samtools/htsjdk/blob/master/src/main/java/htsjdk/samtools/util/TrimmingUtil.java) with one simple method to perform trimming.

Extracting the Mott algorithm from the `TrimmingMethods` will be important for support other kind of trimming once #25 is done, because it will use the same quality enconding.

In addition, trimming Ns could be ported to HTSJDK `TrimmingUtils` to keep our code cleaner, and at some point even the Mott algorithm could be ported.

**Comment by magicDGS** [comment]
`TrimmingUtils` is suppose to be a trimmer for qualities, so the second part (trimming Ns) will not be included. But the Mott algorithm is extracted to a new class in ReadTools in commit 287500b09260800e80a23210b9c6509c4995fb26

**Comment by magicDGS** [comment]
Closing in favor of #32.
